### ISSUE TITLE
.fit() should return self always

### ISSUE THREAD

**Issue by alyakin314** [microsoft/graspologic#361]
while working on #354, realized that inference is not the only place where fit does not return self. 
e.g.:
https://github.com/neurodata/graspy/blob/a3f281c47e717d55a51a6954a88d1fa43ce54def/graspy/match/skp.py#L111
this should probably be standardized throughout graspy.

**Comment by bdpedigo** [comment]
You are not wrong, but that particular class is probably going to be trashed pretty soon. If you see it anywhere else should definitely fix it tho

**Comment by alyakin314** [comment]
okay. I think I looked through everything, and found no such examples, so feel free to close the issue.

do want to note that i have found one code inconsistency. in particular, base class of embed has an implemented (not abstract) fit_transform method, which calls self._fit_transform:
https://github.com/neurodata/graspy/blob/a3f281c47e717d55a51a6954a88d1fa43ce54def/graspy/embed/base.py#L148-L167
some things, say, lse and ase, use it and only implement fit, and rely on a base class' fit_transform, whereas mase and omni have a fit_transform that's just a copy-paste of the base class, except for the docstring:
https://github.com/neurodata/graspy/blob/a3f281c47e717d55a51a6954a88d1fa43ce54def/graspy/embed/mase.py#L215-L234
and mds has it's own overloaded fit_transform:
https://github.com/neurodata/graspy/blob/a3f281c47e717d55a51a6954a88d1fa43ce54def/graspy/embed/mds.py#L209-L232

to clarify: i agree that this does not change the behavior when calling methods, but it may lead to some confusion when changing code. as an example, if someone was to change fit_transform of the base class, it would affect lse and ase, but not mase and omni. whereas if one was to change _fit_transform of the base, it would affect lse, ase, mase and omni. personally, i think it would make most sense for mds add a line of the sort
`self.latent_left_ = self.components_ @ np.diag(self.singular_values_)`
to its fit and let base class' _fit_transform take care of the rest.

and then, either delete fit_transform from all classes but base, or either make the base one abstract, add the implementations everywhere and have them always just 
`return self._fit_transform(input)`
the difference between the two approaches is that the first would mean having the same docstring, but the latter allows to use different ones.

again, not an urgent issue by any means, since it only matters for someone who messes with the source, whether it is us or someone else. it's just a little confusing to have a method that is implemented, sometimes used, sometimes overloaded with a copy of itself, and sometimes overloaded with something else.

the only reason I even looked into it is trying to understand what's the standard way of doing it throughout graspy and found no standard. I am leaning towards the approach with having shared _fit_predict and class-specific fit_predicts with unique docstrings in inference, but happy to listen to you opinions.

**Comment by j1c** [comment]
Yea interesting points. When I wrote MDS, I wanted the attributes to look like those of [sklearn PCA](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html?highlight=pca#sklearn.decomposition.PCA), which is why the attributes are named as they are and why the `fit_transform` is overloaded to accommodate different names. Thats why there's a private and public `fit_transform` function. I see an argument for consistent attribute naming, allowing better factoring of the code, but at the time I wrote it, I thought following PCA was better. Now, I have little opinion on what is better..

As for why I didn't factor `fit_transform` into base class for multigraph method, Im not sure... It makes sense for that to be in base.
